static/app/stores/groupStore.tsx

```diff
@@ -119,7 +119,9 @@ const storeConfig: GroupStoreDefinition = {
   },
 
   mergeItems(items: Item[]) {
-    const itemsById = items.reduce((acc, item) => ({...acc, [item.id]: item}), {});
+    const itemsById = items
+      .filter(item => item && item.id)
+      .reduce((acc, item) => ({...acc, [item.id]: item}), {});
```

Comment:
This can be done in a single pass. Old code is also On^2 due to acc spread

```suggestion
      .reduce((acc, item) => 
          if(!item) return acc;
          acc[item.id] = item;
          return acc;
       ), {});
```